src/index.ts

```diff
@@ -160,6 +190,10 @@ function ViteImageOptimizer(optionsParam: Options = {}): Plugin {
       const files: string[] = getFilesToProcess(allFiles, path => (bundler[path] as any).name);
 
       if (files.length > 0) {
+        if (options.cache === true && !fs.existsSync(options.cacheLocation)) {
```

Comment:
Should we also handle the case where there are no image assets in the `src` directory, but images are only present in the `public` folder?

The `generateBundle` function will run for files present in the `src` directory but not in the `public` directory. We check if image files exist in the `public` directory in the `closeBundle` function.

In that scenario, do you think the creation of the cache folder should also be added in the `closeBundle` call?

Reply:
Good point, updated this.